src/validation.rs

```diff
@@ -223,7 +224,73 @@ impl Validator {
                 operator,
                 ..
             } => {
+                // visit called pou
                 self.visit_statement(operator, context);
+
+                // try to find pou
+                let pou = match operator.as_ref() {
+                    AstStatement::Reference { name, .. } => Some(name),
+                    AstStatement::QualifiedReference { elements, .. } => {
+                        if let Some(stmt) = elements.last() {
+                            if let Some(StatementAnnotation::Variable { resulting_type, .. }) =
+                                context.ast_annotation.get(stmt)
+                            {
+                                Some(resulting_type)
+                            } else {
+                                None
+                            }
+                        } else {
+                            None
+                        }
+                    }
+                    _ => None,
+                }
+                .and_then(|pou_name| context.index.find_pou(pou_name));
+
+                // for PROGRAM/FB we need special inout validation
+                if let Some(PouIndexEntry::FunctionBlock { name, .. })
```

Comment:
So functions have a different validation? I'm not sure what we ended up with. Do functions validate that all arguments are passed?

Reply:
we said for PROGRAMS/FBs we need a validation for inouts because inouts are necessery
for FUNCTIONS we will check if the parameter is ByRef/ByVal
ByVal => create default value if given (currently only inputs)
ByRef => create empty pointer
no validation just creating the missing parameters to pass with the function call

Reply:
We can keep functions with no validation for now